src/core/server/saved_objects/import/check_conflicts.ts

```diff
@@ -0,0 +1,220 @@
+ *  - If this is a `Left` "partial match" result:
+ *     A. If there is a single source and destination match, add the destination to the importIdMap and return the import object, which
+ *        will allow `createSavedObjects` to modify the ID before creating the object (thus ensuring a conflict during).
+ *     B. Otherwise, this is an "ambiguous conflict" result; return an error.
+ */
+export async function checkConflicts(
+  objects: Array<SavedObject<{ title?: string }>>,
+  options: CheckConflictsOptions
+) {
+  // Check each object for possible destination conflicts.
+  const importIds = new Set<string>(objects.map(({ type, id }) => `${type}:${id}`));
+  const checkConflictResults = await Promise.all(
+    objects.map(object => checkConflict(object, importIds, options))
+  );
+
+  // Get a map of all inexact matches that share the same destination(s).
+  const ambiguousConflictSourcesMap = checkConflictResults.filter(isLeft).reduce((acc, cur) => {
+    const key = getAmbiguousConflictSourceKey(cur.value);
+    const value = acc.get(key) ?? [];
+    return acc.set(key, [...value, cur.value.object]);
+  }, new Map<string, Array<SavedObject<{ title?: string }>>>());
+
+  const errors: SavedObjectsImportError[] = [];
+  const filteredObjects: Array<SavedObject<{ title?: string }>> = [];
+  const importIdMap = new Map<string, { id: string }>();
+  checkConflictResults.forEach(result => {
+    if (!isLeft(result)) {
+      filteredObjects.push(result.value);
+      return;
+    }
+    const key = getAmbiguousConflictSourceKey(result.value);
+    const sources = getAmbiguousConflicts(ambiguousConflictSourcesMap.get(key)!);
+    const { object, destinations } = result.value;
+    const { type, id, attributes } = object;
+    if (sources.length === 1 && destinations.length === 1) {
+      // This is a simple "inexact match" result -- a single import object has a single destination conflict.
+      importIdMap.set(`${type}:${id}`, { id: destinations[0].id });
+      filteredObjects.push(object);
+      return;
+    }
+    // This is an ambiguous conflict error, which is one of the following cases:
+    //  - a single import object has 2+ destination conflicts ("ambiguous destination")
+    //  - 2+ import objects have the same single destination conflict ("ambiguous source")
+    //  - 2+ import objects have the same 2+ destination conflicts ("ambiguous source and destination")
+    errors.push({
+      type,
+      id,
+      title: attributes?.title,
+      error: {
+        type: 'ambiguous_conflict',
+        sources,
+        destinations,
+      },
+    });
+  });
+
+  return {
+    errors,
+    filteredObjects,
+    importIdMap,
+  };
+}
+
+/**
+ * Assume that all objects exist in the `retries` map (due to filtering at the beginnning of `resolveSavedObjectsImportErrors`).
+ */
+export async function getImportIdMapForRetries(
```

Comment:
The `checkConflicts` function above is used by `import_saved_objects`, while the `getImportIdMapForRetries` here is used by `resolve_import_errors`.

I decided to split this out and avoid the preflight checks when resolving import errors. The primary reason for this is that there is complex logic in `checkConflicts` to determine what one or more "conflict destinations" will result in -- an inexact match will omit any conflict destinations that exactly match other import objects. However, when resolving import errors, the import objects are filtered out based on the retries. So if you try to import objects A, B, and C, but only B resulted in an inexact match conflict, you will only specify a retry for B when resolving import errors.

My thinking is that we've already given the consumer all of the relevant information regarding inexact match conflicts and ambiguous conflicts, and the consumer will need to make decisions about how to resolve each import error. So the most straightforward / least error-prone thing to do when resolving import errors is obey the retry overwrites without attempting to do any preflight checks.